airflow/utils/timeout.py

```
@@ -39,8 +40,9 @@ def handle_timeout(self, signum, frame):
 
     def __enter__(self):
         try:
-            signal.signal(signal.SIGALRM, self.handle_timeout)
-            signal.alarm(self.seconds)
+            if isinstance(threading.current_thread(), threading._MainThread):
```

If a caller tries to use this outside of the main thread, should we throw an exception?